src/components/navigation/AppNavigation.js

```diff
@@ -0,0 +1,145 @@
+
+const styles = {
+  root: {
+    width: '100%'
+  },
+  flex: {
+    flex: 1,
+    textTransform: 'capitalize'
+  },
+  list: {
+    width: 250,
+    flex: 'initial',
+  },
+  listFull: {
+    width: 'auto',
+    flex: 'initial',
+  }
+};
+
+
+class AppNavigation extends Component {
+  state = {
+    open: false
+  };
+
+  handleOpen = () => this.setState({open: true});
+  handleClose = () => this.setState({open: false});
+  logoutUser = (e) => {
+    this.props.logoutUser();
+    browserHistory.push("/login")
+  }
+
+  render() {
+    const classes = this.props.classes;
+    const publicNav = (
+      <div>
+        <Link to="/login">
+          <ListItem button>
+            <ListItemIcon>
+              <InboxIcon/>
+            </ListItemIcon>
+            <ListItemText primary="Login"/>
+          </ListItem>
+        </Link>
+
+        <Link to="/register">
+          <ListItem button>
+            <ListItemIcon>
+              <InboxIcon/>
+            </ListItemIcon>
+            <ListItemText primary="Register"/>
+          </ListItem>
+        </Link>
+      </div>
+    );
+
+    const authenticatedNav = (
+      <div>
+        <Link to="/dashboard">
+          <ListItem button>
+            <ListItemIcon>
+              <InboxIcon/>
+            </ListItemIcon>
+            <ListItemText primary="Dashboard"/>
+          </ListItem>
+        </Link>
+        <ListItem button onClick={(event) => this.logoutUser(event) }>
+          <ListItemIcon>
+            <InboxIcon/>
+          </ListItemIcon>
+          <ListItemText primary="Logout"/>
+        </ListItem>
+      </div>
+    );
+
+    const sideList = (
+      <div>
+        <List className={classes.list} disablePadding>
+          {
+            this.props.authenticated ? authenticatedNav: publicNav
+          }
+        </List>
+      </div>
+    );
+
+
+    return (
+      <div className={classes.root}>
+        <AppBar position="static">
+          <Toolbar>
+            <IconButton color="contrast" aria-label="Menu">
+              <MenuIcon onClick={this.handleOpen}/>
+            </IconButton>
+            <Typography type="title" color="inherit" className={classes.flex}>
+              {this.props.username ? `Welcome ${this.props.username}` : 'Awesome Bucketlist' }
+            </Typography>
+              {
+                this.props.route === "/register" &&
+                  <Link to="/login"><Button color="contrast">Login</Button></Link>
+              }
+              { this.props.route === "/login" &&
+                <Link to="/register"><Button color="contrast">Sign Up</Button></Link>
+              }
+          </Toolbar>
+        </AppBar>
+        <Drawer open={this.state.open} onRequestClose={this.handleClose} onClick={this.handleClose}>
+          {sideList}
+        </Drawer>
+      </div>
+    );
+  }
+}
+
+AppNavigation.propTypes = {
+  classes: PropTypes.object.isRequired
+};
+
+const mapDispatchToProps = dispatch => ({
```

Comment:
If the `mapDispatchToProps` returns an object whose keys are the same as the action function names, we can forego its definition in favour of passing an object of actions to `connect` in place of `mapDispatchToProps`.
```javascript
connect(null, { logoutUser })(Component)
```